File: php/features.php
<?php
declare(strict_types=1);

// ChineseMedicineClinics SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ChineseMedicineClinicsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ChineseMedicineClinicsBaseFeature();
            case "test":
                return new ChineseMedicineClinicsTestFeature();
            default:
                return new ChineseMedicineClinicsBaseFeature();
        }
    }
}

File: py/features.py
# ChineseMedicineClinics SDK feature factory

from feature.base_feature import ChineseMedicineClinicsBaseFeature
from feature.test_feature import ChineseMedicineClinicsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ChineseMedicineClinicsBaseFeature(),
        "test": lambda: ChineseMedicineClinicsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
